src/meta/meta_ckpt.c

```diff
@@ -131,11 +131,13 @@ __wt_meta_checkpoint(WT_SESSION_IMPL *session, const char *fname, const char *ch
         WT_ERR_NOTFOUND_OK(__wt_config_getones(session, config, "live_restore", &v), true);
         if (ret != WT_NOTFOUND) {
             WT_CONFIG_ITEM cval;
-            WT_ERR_NOTFOUND_OK(__wt_config_subgets(session, &v, "bitmap", &cval), true);
+            WT_ERR_NOTFOUND_OK(__wt_config_subgets(session, &v, "nbits", &cval), true);
             if (ret != WT_NOTFOUND) {
-                WT_ERR(__wt_strndup(session, cval.str, cval.len, &lr_fh_meta->bitmap_str));
-                WT_ERR(__wt_config_subgets(session, &v, "nbits", &cval));
-                lr_fh_meta->nbits = (uint64_t)cval.val;
+                lr_fh_meta->nbits = cval.val;
+                if (lr_fh_meta->nbits > 0) {
+                    WT_ERR(__wt_config_subgets(session, &v, "bitmap", &cval));
+                    WT_ERR(__wt_strndup(session, cval.str, cval.len, &lr_fh_meta->bitmap_str));
+                }
```

Comment:
nit: We could explicitly set `bitmap_str = NULL` in an else branch here. I assume we're depending on the zero initialization to do it for us here.

Reply:
5ce7a597641ccdc272f95b1b4055fe5f827bfd51